pyteal/ast/abi/type.py

```diff
@@ -221,17 +240,23 @@ def store_into(self, output: BaseType) -> Expr:
                 f"expected type_spec {self.produced_type_spec()} but get {output.type_spec()}"
             )
 
-        declaration = self.computation.subroutine.get_declaration()
-
-        if declaration.deferred_expr is None:
-            raise TealInputError(
-                "ABI return subroutine must have deferred_expr to be not-None."
-            )
-        if declaration.deferred_expr.type_of() != output.type_spec().storage_type():
-            raise TealInputError(
-                f"ABI return subroutine deferred_expr is expected to be typed {output.type_spec().storage_type()}, "
-                f"but has type {declaration.deferred_expr.type_of()}."
-            )
+        # HANG NOTE! This get_declaration check applies only for pre frame pointer case
+        # the post frame pointer case should not apply
+        # need to somehow expose the context of evaluation
+        try:
+            declaration = self.computation.subroutine.get_declaration()
```

Comment:
this line is somewhat off: the reason is that, we are not *version aware* when we call `store_into`. This `get_declaration` always return version 6 implementation, while in version 8 this would cause error.

Reply:
For status quo, we cannot know compile to version while evaluating the pyteal source code. We have to use `get_declaration_by_option(False)` here, and we can deprecate `get_declaration` methods.